### ISSUE TITLE
Automating the process of deploying API docs.

### ISSUE THREAD

**Issue by m-murad** [Cadasta/api-docs#40]
Currently if we need to update the `gh-pages`, we need to do it manually by running a script.
We can automate the deployment of API docs to `gh-pages` branch when ever new code is pushed to the `master` branch. This can be done using travis. 

What do you think, Should I send a PR for this?

**Comment by oliverroick** [comment]
Yes, that's a point we've been thinking about as well. Though, I wouldn't say we should deploy `gh-pages` with every push to master; I'd deploy whenever a new tag is created. That allows us to release documentation whenever the changes to the API are released and to keep docs and released API in sync. 

I would suggest to send in a PR and we'll take it from there.

**Comment by m-murad** [comment]
@oliverroick 
Does Cadasta follow any particular pattern while creating a TAG, eg creating tag after every 1 month or n commits. We can trigger a build that will release TAGs for both repositories.

**Comment by oliverroick** [comment]
We would release updates to the API when needed. Smaller fixes would be released whenever ready. When documenting fundamental changes to the API, the docs should be released when the API is released. In other words, for the time being, we would create tags manually, releasing changes according to a schedule is not feasible.

**Comment by m-murad** [comment]
Then i guess that auto updating the `gh-pages` branch when a commit is pushed to `master` branch is the best way to keep the api docs updated. Thats what I have done in the #44

**Comment by oliverroick** [comment]
What I meant was, that we only push to master, when a new version of the API has been tagged. So the process would be to manually tag a version, and Travis runs the deploy only when `$TRAVIS_TAG` is set.
